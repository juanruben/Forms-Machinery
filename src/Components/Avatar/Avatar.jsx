import React from 'react';
import PropTypes from 'prop-types';
import './Avatar.scss';

function Avatar(props) {
    const { image, title } = props;
    return (
        <div className="avatar-container">
            <img src={image} alt="avatar" />
            {title && (
                <div className="title">
                    {`Usuario ${title}`}
                </div>
            )}
        </div>
    );
}

Avatar.propTypes = {
    image: PropTypes.string.isRequired,
    title: PropTypes.string,
};

Avatar.defaultProps = {
    title: null,
};

export default Avatar;
