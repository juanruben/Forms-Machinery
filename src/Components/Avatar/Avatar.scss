.avatar-container {
  margin-bottom: 20px;
  margin-top: 40px;
  text-align: center;

  img {
    margin-bottom: 15px;
  }

  .title {
    font-weight: bold;
  }
}
