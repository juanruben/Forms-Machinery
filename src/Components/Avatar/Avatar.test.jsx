import React from 'react';
import { shallow } from 'enzyme';
import Avatar from './Avatar';

describe('Avatar', () => {
    test('should render ok', () => {
        const wrapper = shallow(<Avatar image="test" title="test" />);
        expect(wrapper.debug()).toMatchSnapshot();
    });
});
