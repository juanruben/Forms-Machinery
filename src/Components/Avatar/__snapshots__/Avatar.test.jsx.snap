// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar should render ok 1`] = `
"<div className=\\"avatar-container\\">
  <img src=\\"test\\" alt=\\"avatar\\" />
  <div className=\\"title\\">
    Usuario test
  </div>
</div>"
`;
